### PR TITLE
Fix library name on Cygwin, MSYS and MinGW when using cmake

### PR DESCRIPTION
**Pull request by mtl1979**
See #971.

### CODE REVIEW

**Comment by nmoinvaz** [comment]
Doesn't this line in madler/zlib mean that it is expecting zlib1.dll for all WIN32?
https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/CMakeLists.txt#L208-211

**Comment by mtl1979** [comment]
@nmoinvaz I'm assuming `UNIX` is defined also when running Cygwin, MSYS and MinGW... All those have own names for the shared library. I can recheck the shared library name on hard disk of my old development PC, but it's too slow to compile code.

**Comment by mtl1979** [comment]
@nmoinvaz I changed the code, not sure if we want filename zlibstatic-ng.lib on Windows when compiling with Visual C++...

Open for suggestions as long as it doesn't cause import library of dll to have same name as static library.

File: CMakeLists.txt [hunk 1]
```diff
@@ -994,7 +994,24 @@ foreach(ZLIB_INSTALL_LIBRARY ${ZLIB_INSTALL_LIBRARIES})
 endforeach()
 
 if(WIN32)
-    set_target_properties(${ZLIB_INSTALL_LIBRARIES} PROPERTIES OUTPUT_NAME zlib${SUFFIX})
+    # Shared library
+    if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+        set_target_properties(zlib PROPERTIES OUTPUT_NAME zlib${SUFFIX})
+    endif()
+    # Static library
+    if(NOT DEFINED BUILD_SHARED_LIBS)
```
Comment (nmoinvaz) [inline CMakeLists.txt:1002]: Can you combine these four `set_target_properties` calls using `${ZLIB_INSTALL_LIBRARIES}` as the target?

**Comment by nmoinvaz** [comment]
I'm not quite convinced that we need this because madler/zlib doesn't use this right? What am I missing?

File: CMakeLists.txt (hunk 1 above)
Comment (mtl1979) [reply CMakeLists.txt:1002]: I don't think so...  `${ZLIB_INSTALL_LIBRARIES}` can also contain the shared library.

I tried combining matching ones using `zlibstatic` as target name but it did throw error that aliases can't be used with `set_target_properties`.

**Comment by mtl1979** [comment]
@nmoinvaz Stock zlib predates Cygwin, MSYS and MinGW, so its `CMakeLists.txt` wasn't fully tested with them. It used separate Makefile that did skip even `configure`.

File: CMakeLists.txt (hunk 1 above)
Comment (nmoinvaz) [reply CMakeLists.txt:1002]: What about using: `z$<$<C_COMPILER_ID:MSVC>:libstatic>${SUFFIX}`

Comment (mtl1979) [reply CMakeLists.txt:1002]: That's pretty much unreadable to average developer...

**Pull request merged**